### PR TITLE
Fix race conditions in manual offset computation for projections

### PR DESCRIPTION
**Pull request by adlerhurst**
Implement manual offset computation to prevent race conditions when events of the same type are written simultaneously. Adjust logging for better debugging and ensure the query avoids using offset to maintain consistency.

### CODE REVIEW

File: internal/eventstore/handler/v2/handler.go [hunk 1]
```diff
@@ -644,7 +680,27 @@ func skipPreviouslyReducedStatements(statements []*Statement, currentState *stat
 			return i
 		}
 	}
-	return -1
+
+	// If no exact match found, look for the last statement at a position less than current
+	// This handles edge cases where exact match isn't found due to race conditions
+	lastValidIndex := -1
```
Comment (adlerhurst) [inline internal/eventstore/handler/v2/handler.go]: FYI @muhlemmer I'll do some additional load tests to verify the correct behavior.

Comment (muhlemmer) [reply internal/eventstore/handler/v2/handler.go]: As I already see load-tests, does that mean that's all done?

**Review by muhlemmer — DISMISSED** [review]
Looks good to me, not sure if @adlerhurst still wants to add additional loadtests.

**Comment by adlerhurst** [comment]
This PR got replaced by https://github.com/zitadel/zitadel/pull/10560
